__tests__/versionCheck.test.js

```diff
@@ -1,40 +1,68 @@
-const {
-  checkCppVersion,
-} = require('../src/reanimated2/platform-specific/checkCppVersion');
-
-const {
-  checkPluginVersion,
-} = require('../src/reanimated2/platform-specific/checkPluginVersion');
-
 const {
   checkVersion,
 } = require('../src/reanimated2/platform-specific/checkVersion');
 const { version: packageVersion } = require('../package.json');
 
 describe('desc', () => {
-  beforeAll(() => {
+  beforeEach(() => {
     global._REANIMATED_VERSION_CPP = packageVersion;
-    global._REANIMATED_VERSION_BABEL_PLUGIN = packageVersion;
-    jest.spyOn(console, 'error');
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    jest.spyOn(console, 'error').mockImplementation(() => {});
   });
 
-  afterAll(() => {
+  afterEach(() => {
     delete global._REANIMATED_VERSION_CPP;
     console.error.mockRestore();
   });
 
-  it('checks cpp version successfully', () => {
-    checkCppVersion();
+  it('checks version successfully', () => {
+    checkVersion();
     expect(console.error).not.toHaveBeenCalled();
   });
 
-  it('checks plugin version successfully', () => {
-    checkPluginVersion();
-    expect(console.error).not.toHaveBeenCalled();
+  it('throws error when version is undefined', () => {
+    delete global._REANIMATED_VERSION_CPP;
+    checkVersion();
+    expect(console.error).toHaveBeenCalled();
   });
 
-  it('checks all versions successfully', () => {
+  it('invalidates major mismatch', () => {
+    global._REANIMATED_VERSION_CPP = packageVersion.replace(
+      /^(\d+).(\d+).(\d+)$/,
+      (match, major, minor, patch) => {
+        console.log(match);
+        return `${major + 1}.${minor}.${patch}`;
+      }
+    );
+    checkVersion(global._REANIMATED_VERSION_CPP);
+    expect(console.error).toHaveBeenCalled();
+  });
+
+  it('invalidates minor mismatch', () => {
+    global._REANIMATED_VERSION_CPP = packageVersion.replace(
+      /^(\d+).(\d+).(\d+)$/,
+      (match, major, minor, patch) => {
+        return `${major}.${minor + 1}.${patch}`;
+      }
+    );
+    checkVersion();
+    expect(console.error).toHaveBeenCalled();
+  });
+
+  it('validates patch mismatch', () => {
+    global._REANIMATED_VERSION_CPP = packageVersion.replace(
+      /^(\d+).(\d+).(\d+)$/,
+      (match, major, minor, patch) => {
+        return `${major}.${minor}.${patch + 1}`;
+      }
+    );
     checkVersion();
     expect(console.error).not.toHaveBeenCalled();
   });
+
+  it('invalidates version post-string', () => {
+    global._REANIMATED_VERSION_CPP = packageVersion + '-post';
+    checkVersion();
+    expect(console.error).toHaveBeenCalled();
+  });
 });
```

Comment:
Honestly, I was rather thinking of some simple Jest unit tests which just import and test `matchVersion` function just like:
```ts
it('...', () => {
  expect(matchVersion('3.0.0', '3.0.1')).toBeTrue();
  // ...
});
```

Reply:
Honestly after I wrote the first test Github Copilot offered me the rest of the them for free (I only had to make minor adjustments).